src/inmanta/user_setup.py

```diff
@@ -0,0 +1,179 @@
+            raise click.ClickException(
+                "Make sure signing configuration is added to the config. See the documentation for details."
+            )
+
+        click.echo(f"{'Authentication signing config: ' : <50}{click.style('found', fg='green')}")
+
+    else:
+        raise click.ClickException("The user setup was aborted as it was not executed locally on the orchestrator")
+
+
+async def get_connection_pool() -> asyncpg.pool.Pool:
+    """
+    connect to a connection pool. It is the responsibility of the caller to close the return connection pool
+    """
+    try:
+        database_host = server_config.db_host.get()
+        database_port = server_config.db_port.get()
+
+        database_username = server_config.db_username.get()
+        database_password = server_config.db_password.get()
+        connection_pool_min_size = server_config.db_connection_pool_min_size.get()
+        connection_pool_max_size = server_config.db_connection_pool_max_size.get()
+        connection_timeout = server_config.db_connection_timeout.get()
+        return await data.connect(
+            database_host,
+            database_port,
+            server_config.db_name.get(),
+            database_username,
+            database_password,
+            create_db_schema=False,
+            connection_pool_min_size=connection_pool_min_size,
+            connection_pool_max_size=connection_pool_max_size,
+            connection_timeout=connection_timeout,
+        )
+    except Exception:
+        raise ConnectionPoolException
+
+
+async def do_user_setup() -> None:
+    """Perform the user setup that requires the database interaction"""
+    connection = None
+    try:
+        click.echo(
+            f"{'Trying to connect to DB:': <50}"
+            f"{('%s (%s:%s)' % ( server_config.db_name.get(), server_config.db_host.get(), server_config.db_port.get()))}"
+        )
+        connection = await get_connection_pool()
+        click.echo(f"{'Connection to database' : <50}" f"{click.style('success', fg='red')}")
+        users = await data.User.get_list()
+
+        if len(users):
+            raise click.ClickException("There are already users in the database")
+
+        username = click.prompt("What username do you want to use?", default="admin")
+        password = click.prompt("What password do you want to use?", hide_input=True)
+
+        pw_hash = nacl.pwhash.str(password.encode())
+
+        # insert the user
+        user = data.User(
+            username=username,
+            password_hash=pw_hash.decode(),
+            auth_method="database",
+        )
+        await user.insert()
+
+        click.echo(f"{'User %s: ' %username : <50}{click.style('created', fg='green')}")
+    except ConnectionPoolException as e:
+        click.echo(f"{'Connection to database {server_config.db_host.get()}' : <50}" f"{click.style('failed', fg='red')}")
+        raise e
+    except UndefinedTableError:
```

Comment:
I think it's better to catch the PostgresError exception here. This is the base class for all PostgreSQL-related errors. If we change the schema of the user table in the future, we might get a different error when the schema is out of date.
```suggestion
    except PostgresError:
```

Reply:
In addition to this, I think you'll either need to use raw postgres queries, or also catch a pydantic error for the `data.User.get_list()`, e.g. if we add a field to the users table but the database has not been upgraded yet.